### PR TITLE
Use furo theme for docs

### PR DESCRIPTION
**Pull request by AVHopp**
This PR introduces the furo theme for creating the docs. It also introduces a few minor fixes like repairing a non-functional link in CONTRIBUTING.md and by ignoring links that are not yet active. Furthermore, the README.md file is now the "Welcome" page, i.e., the first page that is seen when clicking on the documentation.

Note that the coloring scheme is not yet final, and probably the most important aspect to discuss. Further note that this theme has a light and a dark mode.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ make sure that the newly written code is well integrated into the existing packa
 module hierarchy.
 In particular, public functionality should be imported into the appropriate high-level
 namespaces for easier user import. For an example, see our
-[parameter namespace](baybe/parameters/__init__.py).
+[parameter namespace](https://github.com/emdgroup/BayBE/blob/main/baybe/parameters/__init__.py).
```
Comment (Scienfitz) [inline CONTRIBUTING.md:114]: is this now the gerneal recommended and best way to include links into the .md files? Asking because needs to be changed in a lot of README places too (not necessarily this PR)

File: docs/index.md [hunk 2]
```diff
@@ -2,15 +2,18 @@
...
 :titlesonly:
 :caption: General
+:hidden:
 
-
-Overview <self>
 Readme <misc/readme_link>
 User Guide <userguide/userguide>
 Examples <examples/examples>
 How To Contribute <misc/contributing_link>
```
Comment (Scienfitz) [inline docs/index.md:10]: Can we please shorten this to simply `Contribute`, we should keep the side tab as short as possible and no meaningful understanding is lost by this shortening

File: docs/index.md [hunk 3]
```diff
@@ -2,15 +2,18 @@
...
 ```
 
+```{include} ../README.md
+:relative-docs: docs/
+```
+
 # Package documentation
```
Comment (Scienfitz) [inline docs/index.md:17]: Can we please also shorten this to simply `Package` ? I think `Package documentation` unnecessarily sticks out a lot in terms of length from all other topics. At the same time if Im looking for a package code documentation Im pretty sure Id go here if I read only `Package` so we dont loose anything

**Review by AdrianSosic — APPROVED** [review]

**Review by Scienfitz — APPROVED** [review]
